### PR TITLE
added custom user auth and tests

### PR DESCRIPTION
**Pull request by TNAucoin**
- added accounts app
- added new custom user auth models, and related forms
- set new custom user auth model to be default for Django
- added associated unit tests for custom user auth model
- updated .vscode workspace configurations
- added test command to makefile

### CODE REVIEW

File: accounts/admin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    add_form = CustomUserCreationForm
+    form = CustomUserChangeForm
+    model = CustomUser
+    list_display = [
+        "email",
+        "username",
+        "is_superuser",
+    ]
+
+
```
Comment (wdonray) [inline accounts/admin.py:20]: **nitpick:** There is a few extra empty lines in these `py` files just wondering if thats needed or can we format these files.

File: accounts/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                (
+                    "last_name",
+                    models.CharField(
+                        blank=True, max_length=150, verbose_name="last name"
+                    ),
+                ),
+                (
+                    "email",
+                    models.EmailField(
+                        blank=True, max_length=254, verbose_name="email address"
```
Comment (wdonray) [inline accounts/migrations/0001_initial.py:75]: **question:** Is there any other validators we can add to the email?

File: accounts/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                        default=False,
+                        help_text="Designates whether the user can log into this admin site.",
+                        verbose_name="staff status",
+                    ),
+                ),
+                (
+                    "is_active",
+                    models.BooleanField(
+                        default=True,
+                        help_text="Designates whether this user should be treated as active. Unselect this instead of deleting accounts.",
```
Comment (wdonray) [inline accounts/migrations/0001_initial.py:90]: **praise:** Nice `help_text` !

File: accounts/migrations/0001_initial.py (hunk 2 above)
Comment (TNAucoin) [reply accounts/migrations/0001_initial.py:75]: I'm extending default django user class, all that stuff is tucked away in there

File: accounts/admin.py (hunk 1 above)
Comment (TNAucoin) [reply accounts/admin.py:20]: Its a part of the linting, python expects a newline at eof

File: accounts/tests.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+
+# Create your tests here.
```
Comment (wdonray) [inline accounts/tests.py:4]: **nitpick:** I assume this comment can be removed

**praise:**  What a chad for writing test in a personal app.

File: accounts/migrations/0001_initial.py (hunk 3 above)
Comment (TNAucoin) [reply accounts/migrations/0001_initial.py:90]: Thanks, Django did that for me 👍

**Review by wdonray — APPROVED** [review]

File: accounts/tests.py (hunk 4 above)
Comment (TNAucoin) [reply accounts/tests.py:4]: Yes it can

**Pull request merged**